onnxruntime/core/optimizer/skip_layer_norm_fusion.cc

```diff
@@ -217,11 +217,18 @@ Status SkipLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
       continue;
     }
 
+    std::vector<NodeArg*> ln_node_inputs_defs = ln_node.MutableInputDefs();
+
     // Get the inputs for the new SkipLayerNormalization node.
     std::vector<NodeArg*> skip_layer_norm_input_defs{p_add1->MutableInputDefs()[0],
                                                      p_add1->MutableInputDefs()[1],
-                                                     ln_node.MutableInputDefs()[1],
-                                                     ln_node.MutableInputDefs()[2]};
+                                                     ln_node_inputs_defs[1]};
+
+    NodeArg beta_place_holder("", nullptr);
+    if (ln_node_inputs_defs.size() == 2) { // No beta
+      ln_node_inputs_defs.push_back(&beta_place_holder);
+    }
+    skip_layer_norm_input_defs.push_back(ln_node_inputs_defs[2]); // Add Beta or the corresponding placeholder
```

Comment:
Can we push_back(beta_place_holder) so that need not push to vector ln_node_inputs_defs.